test/unittests/libevm/VMTest.cpp

```diff
@@ -860,6 +859,191 @@ class LegacyVMPrecompileCallFixture : public PrecompileCallFixture
 public:
     LegacyVMPrecompileCallFixture() : PrecompileCallFixture{new LegacyVM} {}
 };
+
+class CallFixture : public TestOutputHelperFixture
+{
+public:
+    explicit CallFixture(VMFace* _vm) : vm{_vm}
+    {
+        state.addBalance(myAddress, 2 * ether);
+        state.addBalance(otherAddress, 2 * ether);
+    }
+
+    void testCallCorrectCost(bool _createNewAccount = false, bool _callSelf = false)
+    {
+        assert(!_createNewAccount || !_callSelf);
+
+        // Note: Code passes destination address in via input data. rshift 96 bytes so that correct
+        // 20 bytes are selected during bits -> address conversion in VM (which is BE)
+        //
+        //	let destination : = shr(96, calldataload(0))
+        //  let foo : = call(50000, destination, 1000000000000000000, 0, 0, 0, 0)
+        auto const codeBytes =
+            fromHex("60003560601c6000600060006000670de0b6b3a76400008561c350f15050");
+        auto calleeAddress = otherAddress;
+        if (_createNewAccount)
+            calleeAddress = newAddress;
+        else if (_callSelf)
+            calleeAddress = myAddress;
+        ExtVM extVm{state, envInfo, *se, myAddress, myAddress, myAddress, value, gasPrice,
+            calleeAddress.ref(), ref(codeBytes), sha3(codeBytes), version,
+            static_cast<unsigned int>(depth), isCreate, false /* static call */};
+
+        bigint gasBefore;
+        bigint gasAfter;
+        auto onOp = [&gasBefore, &gasAfter](uint64_t /*steps*/, uint64_t /* PC */,
+                        Instruction _instr, bigint /*newMemSize*/, bigint /*gasCost*/, bigint _gas,
+                        VMFace const*, ExtVMFace const*) {
+            if (CallFixture::IsCallOp(_instr))
+                gasBefore = _gas;
+            else if (gasBefore != 0 && gasAfter == 0)
+                gasAfter = _gas;
+        };
+        vm->exec(gas, extVm, onOp);
+        // Subtract the call stipend gas since there's no code at the address being called
+        unsigned int gasExpected =
+            extVm.evmSchedule().callValueTransferGas - extVm.evmSchedule().callStipend;
```

Comment:
I'm a bit lost about this. Can you explain how it can cost less that transfer gas?

Reply:
@gumb0 Yes - according to the definition of the call stipend (gcallstipend) in the YP, the call stipend is subtracted from the value transfer gas (for a non-zero value transfer). Additionally, the call stipend is added to the gas being passed to the call and its purpose is to ensure that there's just enough gas for the contract's fallback function to perform some basic steps (e.g. log something), but not enough for it to do something nefarious. Also, gas not used in a contract call is refunded to the caller. So, since there's no code at the addresses being called in my test cases, the stipend is refunded.